packages/fiori/cypress/specs/SideNavigation.cy.tsx

```diff
@@ -308,6 +308,62 @@ describe("Side Navigation interaction", () => {
 		});
 	});
 
+	it("Tests link opening with mouse click", () => {
+		cy.mount(
+			<SideNavigation id="sideNav">
+				<SideNavigationItem id="item" text="1" />
+				<SideNavigationItem id="unselectableItemWithLink" text="external link" target="_blank" unselectable={true} href="#test"/>
+				<SideNavigationItem text="3">
+					<SideNavigationSubItem id="SubItemWithLink" text="external link" target="_blank" href="#test"/>
+				</SideNavigationItem>
+			</SideNavigation>
+		);
+
+		cy.url().should("not.include", "#test");
+		cy.get("#unselectableItemWithLink").shadow().find("a").invoke("removeAttr", "target");
+
+		cy.get("#unselectableItemWithLink").realClick();
+
+		cy.url().should("include", "#test");
+
+		// Remove #test from the URL
+		cy.window().then(win => {
+			win.history.pushState({}, "", win.location.pathname + win.location.search);
+		});
+
+		cy.url().should("not.include", "#test");
+	});
+
+	it("Tests link opening with Enter", () => {
+		cy.mount(
+			<SideNavigation id="sideNav">
+				<SideNavigationItem id="item" text="1" />
+				<SideNavigationItem id="unselectableItemWithLink" text="external link" target="_blank" unselectable={true} href="#test"/>
+				<SideNavigationItem text="3">
+					<SideNavigationSubItem id="SubItemWithLink" text="external link" target="_blank" href="#test"/>
+				</SideNavigationItem>
+			</SideNavigation>
+		);
+
+		cy.url().should("not.include", "#test");
+		cy.get("#unselectableItemWithLink").shadow().find("a").invoke("removeAttr", "target");
+
+		// cy.get("#unselectableItemWithLink").trigger("keydown", { key: "Enter" });
+		cy.get("#unselectableItemWithLink").realPress("Enter");
+
+		cy.url().should("include", "#test");
+
+		// Remove #test from the URL
+		cy.window().then(win => {
+			win.history.pushState({}, "", win.location.pathname + win.location.search);
+		});
+
+		cy.url().should("not.include", "#test");
+	});
+
+	// it("Tests link opening with Space", () => {
+	// });
+
```

Comment:
Implement this test. Use `keyup`. You can use this for reference https://github.com/SAP/ui5-webcomponents/pull/10699/files

Reply:
realPress will trigger keydown and then keyup. why not use it?

Reply:
If I copy the previous test (for Enter) and change it for Space if fails without the fix and passes with the fix.
Do we need to test explicitly the keyup?

Reply:
Yes, please explicitly test with `keyup`. We often make a mistake where we trigger actions on Space keydown. If we test with `realPress`, we risk to not catch such a mistake.